### ISSUE TITLE
Pytest issue after python update to 3.8.10

### ISSUE THREAD

**Issue by viktor-zinchenko** [pytest-dev/pytest#8642]
After python from update from 3.8.9 to 3.8.10 got and issue with pytest.

## Steps to reproduce
1. Install [Faker](https://github.com/joke2k/faker) and pytest as a project dependencies.
1. Run: pytest

Traceback:

```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 415, in get_importer
    importer = sys.path_importer_cache[path_item]
KeyError: PosixPath('/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/providers')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/user/Documents/work/work/venv/project/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 185, in console_main
    code = main()
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 143, in main
    config = _prepareconfig(args, plugins)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 318, in _prepareconfig
    config = pluginmanager.hook.pytest_cmdline_parse(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/hooks.py", line 286, in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 93, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 84, in <lambda>
    self._inner_hookexec = lambda hook, methods, kwargs: hook.multicall(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 203, in _multicall
    gen.send(outcome)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/helpconfig.py", line 100, in pytest_cmdline_parse
    config: Config = outcome.get_result()
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 80, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/callers.py", line 187, in _multicall
    res = hook_impl.function(*args)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1003, in pytest_cmdline_parse
    self.parse(args)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1283, in parse
    self._preparse(args, addopts=addopts)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/config/__init__.py", line 1172, in _preparse
    self.pluginmanager.load_setuptools_entrypoints("pytest11")
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 961, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1014, in _gcd_import
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/__init__.py", line 1, in <module>
    from faker.factory import Factory
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/factory.py", line 7, in <module>
    from faker.config import AVAILABLE_LOCALES, DEFAULT_LOCALE, PROVIDERS
  File "<frozen importlib._bootstrap>", line 991, in _find_and_load
  File "<frozen importlib._bootstrap>", line 975, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 671, in _load_unlocked
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/_pytest/assertion/rewrite.py", line 170, in exec_module
    exec(co, module.__dict__)
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/config.py", line 11, in <module>
    PROVIDERS = find_available_providers(
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 57, in find_available_providers
    for mod in list_module(providers_mod) if mod != '__pycache__'
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 35, in list_module
    return [name for _, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
  File "/Users/user/Documents/work/work/venv/project/lib/python3.8/site-packages/faker/utils/loading.py", line 35, in <listcomp>
    return [name for _, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 129, in iter_modules
    for i in importers:
  File "/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/pkgutil.py", line 419, in get_importer
    importer = path_hook(path_item)
  File "<frozen importlib._bootstrap_external>", line 1594, in path_hook_for_FileFinder
  File "<frozen importlib._bootstrap_external>", line 1469, in __init__
  File "<frozen importlib._bootstrap_external>", line 177, in _path_isabs
AttributeError: 'PosixPath' object has no attribute 'startswith'
make: *** [pytest] Error 1
```
## pip list

```
Package                 Version    Location
----------------------- ---------- --------------------------
anybadge                1.7.0
appdirs                 1.4.4
astunparse              1.6.3
attrs                   20.3.0
aws-lambda-powertools   1.14.0
aws-xray-sdk            2.6.0
awscli                  1.19.64
backoff                 1.10.0
bandit                  1.7.0
black                   21.4b2
blessed                 1.17.6
boto3                   1.17.64
botocore                1.20.64
bump2version            1.0.1
bumpversion             0.6.0
certifi                 2020.12.5
chalice                 1.22.1
chardet                 4.0.0
click                   7.1.2
colorama                0.4.3
cosmic-ray              7.0.0
coverage                5.5
coveralls               3.0.1
decorator               5.0.7
docopt                  0.6.2
docutils                0.15.2
enum-compat             0.0.3
exit-codes              1.3.0
Faker                   6.6.3
fastjsonschema          2.15.1
flake8                  3.9.1
flake8-bandit           2.1.2
flake8-bugbear          21.4.3
flake8-docstrings       1.6.0
flake8-logging-format   0.6.0
flake8-polyfill         1.0.2
future                  0.18.2
gitdb                   4.0.7
GitPython               3.1.14
idna                    3.1
iniconfig               1.1.1
inquirer                2.7.0
isort                   5.8.0
iterfzf                 0.5.0.20.0
Jinja2                  2.11.3
jmespath                0.10.0
jsonpickle              2.0.0
mando                   0.6.4
MarkupSafe              1.1.1
marshmallow             3.11.1
mccabe                  0.6.1
mypy                    0.812
mypy-extensions         0.4.3
packaging               20.9
parso                   0.8.2
pathlib                 1.0.1
pathspec                0.8.1
pbr                     5.6.0
peewee                  3.14.4
pip                     19.3
pluggy                  0.13.1
publicsuffix2           2.20191221
py                      1.10.0
pyasn1                  0.4.8
pycodestyle             2.7.0
pydocstyle              6.0.0
pyflakes                2.3.1
PyMySQL                 1.0.2
pyparsing               2.4.7
pytest                  6.2.3
pytest-asyncio          0.14.0
pytest-cov              2.11.1
python-dateutil         2.8.1
python-dotenv           0.17.1
python-editor           1.0.4
pytz                    2021.1
PyYAML                  5.4.1
qprompt                 0.16.1
radon                   4.5.0
readchar                2.0.1
regex                   2021.4.4
requests                2.25.1
responses               0.13.3
rsa                     4.7.2
s3transfer              0.4.2
setuptools              56.1.0
six                     1.16.0
smmap                   4.0.0
snowballstemmer         2.1.0
stevedore               3.3.0
text-unidecode          1.3
toml                    0.10.2
typed-ast               1.4.3
typing-extensions       3.10.0.0
urllib3                 1.26.4
virtualenv              16.7.10
wcwidth                 0.2.5
wheel                   0.36.2
wrapt                   1.12.1
yattag                  1.14.0
```

## Versions
* OS: MacOs 10.15.4
* Python version: 3.8.10

FIY: Caused by this [commit](https://github.com/python/cpython/commit/eed7686e9fe22a3eb5e1a1fc7d27c27fca070bd1) in python release v3.8.10. Compiled python v3.8.10 without it and bug was not reproduced.

**Comment by skwashd** [comment]
There is an issue in the faker queue over at joke2k/faker#1438.

**Comment by The-Compiler** [comment]
This isn't a pytest issue, and apparently has been fixed in faker v7.0.1 (from March).
